### PR TITLE
Kraft

### PR DESCRIPTION
**Pull request by anuj-apdev**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: molecule/broker-scale-up/molecule.yml [hunk 1]
```diff
@@ -94,18 +94,6 @@ platforms:
     privileged: true
     networks:
       - name: confluent
```
Comment (mansisinha) [inline molecule/broker-scale-up/molecule.yml:96]: why are we removing connect2 and ksql2?

File: roles/variables/vars/main.yml [hunk 2]
```diff
@@ -155,11 +155,7 @@ kafka_controller_properties:
       listener.name.controller.plain.sasl.jaas.config: "{{plain_jaas_config}}"
       sasl.mechanism.controller.protocol: PLAIN
   ssl:
-    enabled: "{{
-    kafka_controller_listeners | confluent.platform.ssl_required(kafka_controller_ssl_enabled) or
-    mds_broker_listener.ssl_enabled|bool or
-    mds_tls_enabled|bool or
-    ( kafka_controller_schema_validation_enabled and 'schema_registry' in groups and schema_registry_ssl_enabled ) }}"
+    enabled: "{{ kafka_controller_listeners | confluent.platform.ssl_required(kafka_controller_ssl_enabled) }}"
```
Comment (mansisinha) [inline roles/variables/vars/main.yml:158]: won't we need the other conditions once mds is supported?

File: roles/variables/vars/main.yml [hunk 3]
```diff
@@ -228,11 +202,11 @@ kafka_controller_properties:
     enabled: "{{ kafka_controller_metrics_reporter_enabled|bool }}"
     properties:
       metric.reporters: io.confluent.metrics.reporter.ConfluentMetricsReporter
-      confluent.metrics.reporter.bootstrap.servers: "{{ groups['kafka_controller'] | default(['localhost']) | confluent.platform.resolve_hostnames(hostvars) | join(':' + kafka_controller_listeners['controller']['port']|string + ',') }}:{{kafka_controller_listeners['controller']['port']}}"
-      confluent.metrics.reporter.topic.replicas: "{{kafka_controller_default_internal_replication_factor}}"
+      confluent.metrics.reporter.bootstrap.servers: "{{ groups['kafka_broker'] | default(['localhost']) | confluent.platform.resolve_hostnames(hostvars) | join(':' + kafka_broker_listeners['internal']['port']|string + ',') }}:{{kafka_broker_listeners['internal']['port']}}"
```
Comment (mansisinha) [inline roles/variables/vars/main.yml:205]: why are we using broker here?

Comment (anuj-apdev) [reply roles/variables/vars/main.yml:205]: The controller will report its metrics to the broker. As Ron mentioned, the controller does not have/hosts any topic where we could send/safe metrics.

File: molecule/broker-scale-up/molecule.yml (hunk 1 above)
Comment (anuj-apdev) [reply molecule/broker-scale-up/molecule.yml:96]: Having 5 brokers made this scenario very heavy, and was failing many times due to less memory on the system. Anyways this is a broker-scale-up scenario, so there was not much importance in having >1 `ksql` and `connect`. We already have scenarios for `ksql` and connect scale-up.

File: roles/variables/vars/main.yml (hunk 2 above)
Comment (anuj-apdev) [reply roles/variables/vars/main.yml:158]: yeah, you are right. I wrote this code when I was under the impression that `mds` would not be supported in the near future.
But that turned up not very accurate.
But in either case, `mds` service would be running on the broker and not on the controller. So this condition will not be used as it is even in the future.

**Review by mansisinha — APPROVED** [review]

**Pull request merged**